keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClient.java

```diff
@@ -0,0 +1,593 @@
+     *       System.out.printf("Key with name %s and value %s \n", keyResponse.value().name(), keyResponse.value().id())));
+     * </pre>
+     *
+     * @param keyBase The {@link KeyBase base key} holding attributes of the key being requested.
+     * @throws ResourceNotFoundException when a key with {@link KeyBase#name() name} and {@link KeyBase#version() version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@link KeyBase#name()}  name} or {@link KeyBase#version() version} are empty strings.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the requested {@link Key key}.
+     */
+    public Mono<Response<Key>> getKey(KeyBase keyBase) {
+        Objects.requireNonNull(keyBase, "The Key Base parameter cannot be null.");
+        String keyVersion = "";
+        return service.getKey(endpoint, keyBase.name(), keyVersion, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * Updates the attributes associated with the specified key, but not the cryptographic key material of the specified key in the key vault. The update
+     * operation changes specified attributes of an existing stored key and attributes that are not specified in the request are left unchanged.
+     * The cryptographic key material of a key itself cannot be changed. This operation requires the {@code keys/set} permission.
+     *
+     * <p>The {@code key} is required and its fields {@link KeyBase#name() name} and {@link KeyBase#version() version} cannot be null.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Gets latest version of the key, changes its notBefore time and then updates it in the Azure Key Vault. Subscribes to the call asynchronously and prints out the
+     * returned key details when the callback happens.</p>
+     * <pre>
+     * keyAsyncClient.getKey("keyName").subscribe(keyResponse -&gt; {
+     *     Key key = keyResponse.value();
+     *     //Update the not before time of the key.
+     *     key.notBefore(OffsetDateTime.now().plusDays(50));
+     *     keyAsyncClient.updateKey(key).subscribe(updatedKeyResponse -&gt;
+     *         System.out.printf("Key's updated not before time %s \n", updatedKeyResponse.value().notBefore().toString()));
+     *   });
+     * </pre>
+     *
+     * @param key The {@link KeyBase base key} object with updated properties.
+     * @throws NullPointerException if {@code key} is {@code null}.
+     * @throws ResourceNotFoundException when a key with {@link KeyBase#name() name} and {@link KeyBase#version() version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@link KeyBase#name() name} or {@link KeyBase#version() version} are empty strings.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link KeyBase updated key}.
+     */
+    public Mono<Response<Key>> updateKey(KeyBase key) {
```

Comment:
Need to add an overload which also takes keyOps

Reply:
Currently KeyBase contains KeyOperations as a convenience method, which can be used to set the key operations and update them via update method.
Should we do that or add an overload which takes in key ops?

Reply:
Added the overload to take in keyOps.